source/AMDSBufferGroup.cpp

```diff
@@ -114,6 +119,20 @@ void AMDSBufferGroup::populateData(AMDSClientStartTimeToEndTimeDataRequest* clie
 	}
 }
 
+void AMDSBufferGroup::populateData(AMDSClientMiddleTimePlusCountBeforeAndAfterDataRequest* clientDataRequest)
+{
+	QDateTime middleTime = clientDataRequest->middleTime();
+	int countBefore = clientDataRequest->countBefore();
+	int countAfter = clientDataRequest->countAfter();
+
+	int middleIndex = lowerBound(middleTime);
```

Comment:
I don't really like the name `lowerBound()`.  It doesn't convey a lot of information.  For instance, what is an `int` that is returned a lower bound of a QDateTime?